packages/core/src/dialects/mssql/query-generator-typescript.ts

```diff
@@ -83,4 +85,16 @@ export class MsSqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
       this.quoteTable(tableName),
     ]);
   }
+
+  jsonPathExtractionQuery(sqlExpression: string, path: ReadonlyArray<number | string>, unquote: boolean): string {
+    if (!unquote) {
+      throw new Error(`JSON Paths are not supported in ${this.dialect.name} without unquoting the JSON value.`);
+    }
+
+    return `JSON_VALUE(${sqlExpression}, ${this.escape(buildJsonPath(path))})`;
+  }
+
+  formatUnquoteJson(arg: Expression, options?: EscapeOptions) {
+    return this.escape(arg, options);
```

Comment:
This is incorrect. It needs to unquote, so it needs to use `JSON_VALUE` with a path that returns the root.

Reply:
Thanks for spotting this, I've updated it now